Sources/PointFree/Environment.swift

```diff
@@ -1,69 +1,168 @@
-    self.calendar = calendar
-    self.cookieTransform = cookieTransform
-    self.collections = collections
-    self.database = database
-    self.date = date
-    self.envVars = envVars
-    self.episodes = episodes
-    self.features = features
-    self.gitHub = gitHub
-    self.logger = logger
-    self.mailgun = mailgun
-    self.renderHtml = renderHtml
-    self.renderXml = renderXml
-    self.stripe = stripe
-    self.uuid = uuid
+import Stripe 
+
+// NB: Deprecate remove soon: @available(*, deprecated)
+public var Current: DependencyValues {
+  DependencyValues._current
+}
+
+extension Logger: DependencyKey {
+  public static let liveValue = Logger(label: "co.pointfree")
+  public static let testValue = Logger(label: "co.pointfree.PointFreeTestSupport")
+}
+
+extension DependencyValues {
+  public var logger: Logger {
+    get { self[Logger.self] }
+    set { self[Logger.self] = newValue }
+  }
+}
+
+extension BlogPost: DependencyKey {
+  public static let liveValue: () -> [BlogPost] = allBlogPosts
+}
+
+extension Episode: DependencyKey {
+  public static var liveValue: () -> [Episode] {
+#if !OSS
+    Episode.bootstrapPrivateEpisodes()
+#endif
+    assert(Episode.all.count == Set(Episode.all.map(\.id)).count)
+    assert(Episode.all.count == Set(Episode.all.map(\.sequence)).count)
+
+    return {
+      @Dependency(\.date.now) var now
+      @Dependency(\.envVars.appEnv) var appEnv
+      return Episode.all
+        .filter {
+          appEnv == .production
+          ? $0.publishedAt <= now
+          : true
+        }
+        .sorted(by: their(\.sequence))
+    }
+  }
+}
+
+private enum RenderHTML: DependencyKey {
+  static let liveValue: (Node) -> String = { Html.render($0) }
+  static let testValue: (Node) -> String = { debugRender($0) }
+}
+
+extension DependencyValues {
+  public var renderHtml: (Node) -> String {
+    get { self[RenderHTML.self] }
+    set { self[RenderHTML.self] = newValue }
+  }
+}
+
+private enum RenderXML: DependencyKey {
+  static let liveValue: (Node) -> String = { Html._xmlRender($0) }
+  static let testValue: (Node) -> String = { _debugXmlRender($0) }
+}
+
+extension DependencyValues {
+  public var renderXml: (Node) -> String {
+    get { self[RenderXML.self] }
+    set { self[RenderXML.self] = newValue }
+  }
+}
+
+private enum MainEventLoopGroupKey: DependencyKey {
+  static var liveValue: MultiThreadedEventLoopGroup {
+#if DEBUG
+    let numberOfThreads = 1
+#else
+    let numberOfThreads = System.coreCount
+#endif
+    return MultiThreadedEventLoopGroup(numberOfThreads: numberOfThreads)
+  }
+}
+
+extension DependencyValues {
+  public var mainEventLoopGroup: MultiThreadedEventLoopGroup {
+    get { self[MainEventLoopGroupKey.self] }
+    set { self[MainEventLoopGroupKey.self] = newValue }
+  }
+}
+
+extension Database.Client: DependencyKey {
+  public static var liveValue: Self {
+    @Dependency(\.envVars) var envVars
+    @Dependency(\.mainEventLoopGroup) var mainEventLoopGroup
```

Comment:
example of using `@Dependency` from within the `liveValue` of another dependency. Seems to work well AFAICT.

Reply:
What happens if you create a circular dependency between `liveValue`s?

Reply:
Makes sense. That's about what I figured. I'm excited to see the implementation of `swift-dependencies`. It looks like a great tool from what I see in the various public `swift-deps` branches 😄